accelerator/proxy_apis.h

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2018-2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef ACCELERATOR_PROXY_APIS_H_
+#define ACCELERATOR_PROXY_APIS_H_
+
+#include "accelerator/errors.h"
+#include "cclient/api/core/core_api.h"
+#include "cclient/request/requests.h"
+#include "cclient/response/responses.h"
+#include "utils/logger_helper.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file proxy_apis.h
+ * @brief Implement Proxy APIs
+ *
+ * tangle-accelerator provides major IRI Proxy APIs wrapper.
+ * The arguments and return strings are all in json format. There can be
+ * different server or protocol integration with these APIs.
+ */
+
+/**
+ * Initialize logger
+ */
+void proxy_apis_logger_init();
+
+/**
+ * Release logger
+ *
+ * @return
+ * - zero on success
+ * - EXIT_FAILURE on error
+ */
+int proxy_apis_logger_release();
+
+/**
+ * Initialize lock
+ *
+ * @return
+ * - zero on success
+ * - SC_CONF_LOCK_INIT on error
+ */
+status_t proxy_apis_lock_init();
+
+/**
+ * Destroy lock
+ *
+ * @return
+ * - zero on success
+ * - SC_CONF_LOCK_DESTROY on error
+ */
+status_t proxy_apis_lock_destroy();
+
+/**
+ * @brief Store response message to json_result
+ * @param[in] res_buff Buffer of response message
+ * @param[out] json_result Response message
+ *
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
+ */
+status_t char_buffer_to_str(char_buffer_t* res_buff, char** json_result);
```

Comment:
Is it necessary to expose this function in public header?

Reply:
I think it's not necessary to expose this function in public header, I will remove it from public header.